src/SprykerEco/Zed/Computop/Persistence/ComputopRepository.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Computop\Persistence;
+
+use Generated\Shared\Transfer\ComputopPaymentComputopDetailTransfer;
+use Generated\Shared\Transfer\ComputopPaymentComputopOrderItemCollectionTransfer;
+use Generated\Shared\Transfer\ComputopPaymentComputopTransfer;
+use Generated\Shared\Transfer\ComputopSalesOrderItemCollectionTransfer;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+
+/**
+ * @method \SprykerEco\Zed\Computop\Persistence\ComputopPersistenceFactory getFactory();
+ */
+class ComputopRepository extends AbstractRepository implements ComputopRepositoryInterface
+{
+    /**
+     * @param string $transactionId
+     *
+     * @return \Generated\Shared\Transfer\ComputopPaymentComputopTransfer
+     */
+    public function getComputopPaymentByComputopTransId(string $transactionId): ComputopPaymentComputopTransfer
+    {
+        $computopPaymentEntity = $this->getFactory()
+            ->createPaymentComputopQuery()
+            ->filterByTransId($transactionId)
+            ->findOne();
+
+        if ($computopPaymentEntity === null) {
+            return new ComputopPaymentComputopTransfer();
```

Comment:
this is a bit confusing that we return an empty transfer.

WIthout checking this methods body, when I get a transfer I would assume that this is a valid transfer with data